tls/extensions/s2n_server_key_share.c

```diff
@@ -199,21 +201,28 @@ static int s2n_server_key_share_recv_pq_hybrid(struct s2n_connection *conn, uint
     POSIX_ENSURE(client_kem_group_params->ecc_params.evp_pkey, S2N_ERR_BAD_KEY_SHARE);
     POSIX_ENSURE(client_kem_group_params->kem_group == server_kem_group_params->kem_group, S2N_ERR_BAD_KEY_SHARE);
 
-    uint16_t received_total_share_size;
-    POSIX_GUARD(s2n_stuffer_read_uint16(extension, &received_total_share_size));
-    POSIX_ENSURE(received_total_share_size == server_kem_group_params->kem_group->server_share_size, S2N_ERR_BAD_KEY_SHARE);
-    POSIX_ENSURE(s2n_stuffer_data_available(extension) == received_total_share_size, S2N_ERR_BAD_KEY_SHARE);
+    uint16_t actual_hybrid_share_size = 0;
+    POSIX_GUARD(s2n_stuffer_read_uint16(extension, &actual_hybrid_share_size));
+    POSIX_ENSURE(s2n_stuffer_data_available(extension) == actual_hybrid_share_size, S2N_ERR_BAD_KEY_SHARE);
+
+    struct s2n_kem_params *client_kem_params = &conn->kex_params.client_kem_group_params.kem_params;
+    POSIX_GUARD(s2n_is_tls13_hybrid_kem_length_prefixed(S2N_SERVER, actual_hybrid_share_size, server_kem_group_params->kem_group, &client_kem_params->len_prefixed));
```

Comment:
See other comments about the `mode` argument. Here is the client calling this function with `mode = S2N_SERVER`. At the very least, I think the argument needs a more descriptive name than simply `mode`. But I don't see why the client can't just check it's own local `kem_params` to determine whether it sent a length prefixed key share in the first place.